api/helpers/gamestate/moves/jack/validate.js

```diff
@@ -0,0 +1,80 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to play Jack',
+
+  description: 'Verifies whether a request to play a Jack is legal, throwing explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'Object containing the current gameState',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to play Jack
+     * @param { String } requestedMove.cardId - Card Played (Jack)
+     * @param { String } [ requestedMove.targetId ] - Card targeted by Jack
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object containing data needed for current move',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting move',
+      required: true,
+    },
+  },
+  sync: true,
+
+  fn: ({ currentState, requestedMove, playedBy }, exits) => {
+    try {
+      const player = playedBy ? currentState.p1 : currentState.p0;
+      const opponent = playedBy ? currentState.p0 : currentState.p1;
+
+      const cardPlayed = player.hand.find(({ id }) => id === requestedMove.cardId);
```

Comment:
Let's rename this variable to `playedCard` to be more consistent with the `GameStateRow`, `GameState` and the `execute()` helpers. I had this backwards in an early `validate()` example and it propogated to all the validate helpers. I'm changing all the others back to `playedCard` in #1051. Let's follow suit here
```suggestion
      const playedCard = player.hand.find(({ id }) => id === requestedMove.cardId);
```